### PR TITLE
window executable available with embeddable python

### PR DESCRIPTION
**Pull request by Arslan-Siraj**
### **User description**
- added .md files for both pyinstaller/embeddable python windows packaging
- added in the build_app documentation
- write test for executables with embeddable python in github actions 
-  write full app (OpenMS app artifacts) in executable: now default exwcutable build by embeddable python

___

### **PR Type**
enhancement, documentation

___

### **Description**
- Added and updated GitHub Actions workflows for building and testing Windows executables using PyInstaller and embeddable Python.
- Updated the Streamlit packaging instructions in the documentation to include new methods using PyInstaller and embeddable Python.
- Introduced new documentation files for creating Windows executables, detailing step-by-step processes.
- Updated Streamlit version in requirements.txt to ensure compatibility with new executable creation methods.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>2_📖_Build_App.py</strong><dd><code>Update Streamlit App Packaging Instructions for Windows</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
pages/2_📖_Build_App.py

<li>Removed outdated instructions for Windows executable package setup.<br> <li> Added a new section on how to package Streamlit apps for Windows using <br>PyInstaller and embeddable Python.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/streamlit-template/pull/57/files#diff-d72e179aaaf7de1792491e646fb37406daf2f2d87299403936aa89fbda5f1a36">+8/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>build-windows-executable-app-with-pyinstaller.yaml</strong><dd><code>Add GitHub Actions Workflow for Building Windows Executable with </code><br><code>PyInstaller</code></dd></summary>
<hr>
      
.github/workflows/build-windows-executable-app-with-pyinstaller.yaml

<li>Added a new GitHub Actions workflow to build a Windows executable <br>using PyInstaller.<br> <li> Detailed steps include environment setup, building with CMake, and <br>packaging with PyInstaller.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/streamlit-template/pull/57/files#diff-02740fb40ac476b44a312979c501dcdd042c91d20cd21389ee0fb446f96bb07c">+253/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>build-windows-executable-app.yaml</strong><dd><code>Modify Workflow to Use Python Embeddable Version for Windows </code><br><code>Executable</code></dd></summary>
<hr>
      
.github/workflows/build-windows-executable-app.yaml

<li>Modified the existing GitHub Actions workflow to use Python's <br>embeddable version for building Windows executables.<br> <li> Added steps for setting up Python, installing pip, and creating an <br>all-in-one executable folder.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/streamlit-template/pull/57/files#diff-211d58f377673fc6e00e49835e1cf5dc727c333368e2b5f8f1cf31e1eff077c5">+36/-53</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>test-win-exe-w-pyinstaller.yaml</strong><dd><code>Update Workflow Name for Clarity in Testing with PyInstaller</code></dd></summary>
<hr>
      
.github/workflows/test-win-exe-w-pyinstaller.yaml

- Updated the workflow name to specify testing with PyInstaller.

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/streamlit-template/pull/57/files#diff-dc62684cbeece9dbcd1bb5a389429f6aac3f97cef626abf07a013f8a4ccab898">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>requirements.txt</strong><dd><code>Update Streamlit Version in Requirements</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
requirements.txt

- Updated Streamlit version from 1.29.0 to 1.33.0.

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/streamlit-template/pull/57/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test-win-exe-w-embed-py.yaml</strong><dd><code>New GitHub Actions Workflow for Testing Windows Executable with </code><br><code>Embeddable Python</code></dd></summary>
<hr>
      
.github/workflows/test-win-exe-w-embed-py.yaml

<li>Created a new GitHub Actions workflow to test the Streamlit executable <br>with embeddable Python on Windows.<br> <li> Includes steps for setting up Python, installing required packages, <br>and packaging the application.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/streamlit-template/pull/57/files#diff-6e75dcaae7958d86488924614c721d7b77f1301fe0b4f79cb982a8481fe8234f">+61/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>win_exe_with_embed_py.md</strong><dd><code>Document Process for Creating Windows Executable with Embeddable </code><br><code>Python</code></dd></summary>
<hr>
      
win_exe_with_embed_py.md

<li>Added a new markdown document detailing the process of creating a <br>Windows executable using embeddable Python.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/streamlit-template/pull/57/files#diff-86653ad0b4ed7fac2ddfc42fd2714d731187b40bc21ca0fd0fb3e39f717ee14a">+110/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>win_exe_with_pyinstaller.md</strong><dd><code>Update Documentation for Creating Windows Executable with PyInstaller</code></dd></summary>
<hr>
      
win_exe_with_pyinstaller.md

<li>Updated the document to reflect support for newer versions of <br>Streamlit and added detailed steps for creating executables with <br>PyInstaller.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/streamlit-template/pull/57/files#diff-81395b482c2e66d24a9db763bedcafb57f571707373ab76dca1ac6fc3ad5c63d">+35/-16</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: pages/2_📖_Build_App.py [hunk 1]
```diff
@@ -79,5 +75,11 @@
 ### Complex workflow using TOPP tools
 
 This template app features a module in `src/workflow` that allows for complex and long workflows to be built very efficiently. Check out the `TOPP Workflow Framework` page for more information (on the *sidebar*).
-""")
 
+## How to package everything for window executables
```
Comment (axelwalter) [inline pages/2_📖_Build_App.py:79]: Could be a separate page in the documentation, just for Windows executables?!
Import markdown files and display content directly in streamlit app [like here?](https://github.com/OpenMS/streamlit-template/blob/main/pages/4_%F0%9F%93%96_Deployment.py)

File: requirements.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # the requirements.txt file is intended for deployment on streamlit cloud and if the simple container is built
 # note that it is much more restricted in terms of installing third-parties / etc.
 # preferably use the batteries included or simple docker file for local hosting
-streamlit==1.29.0
+streamlit==1.33.0
```
Comment (axelwalter) [inline requirements.txt:4]: The latest is now 1.34, are there issues with that version?

File: win_exe_with_embed_py.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+```bash
+./python-3.11.9/python -m pip install -r requirements.txt --no-warn-script-location
+```
+
+### Test and create `run_app.bat` file
+
+1. Test by running app
+
+    ```batch
+        .\python-3.11.9\python -m streamlit run app.py local
```
Comment (axelwalter) [inline win_exe_with_embed_py.md:69]: Is it necessary to distinguish between local mode and Windows executable mode?

**Comment by axelwalter** [comment]
Thanks for the great work @Arslan-Siraj

File: win_exe_with_embed_py.md (hunk 3 above)
Comment (Arslan-Siraj) [reply win_exe_with_embed_py.md:69]: that coming, in the next iteration (work on different mode), for the time being, it is like before.

File: requirements.txt (hunk 2 above)
Comment (Arslan-Siraj) [reply requirements.txt:4]: no issues with lates

File: pages/2_📖_Build_App.py (hunk 1 above)
Comment (Arslan-Siraj) [reply pages/2_📖_Build_App.py:79]: yes, sure it is also possible.

File: win_exe_with_embed_py.md (hunk 3 above)
Comment (Arslan-Siraj) [reply win_exe_with_embed_py.md:69]: https://github.com/OpenMS/streamlit-template/issues/51

File: pages/2_📖_Build_App.py (hunk 1 above)
Comment (Arslan-Siraj) [reply pages/2_📖_Build_App.py:79]: @axelwalter 
Screenshot 2024-05-08 145106
any suggestion here?

**Pull request merged**